### PR TITLE
Derived beacon data

### PR DESCRIPTION
**Pull request by zackads**
## Context

Handle derived beacon data in the back office application.

## Changes in this pull request

- Adds to-be-derived beacon type to heading:

Beacon type to heading - wireframe (Wireframe)

image (As implemented)

- Doesn't allow user to edit derived fields:

image (Wireframe)

image(As implemented)

- Adds derived fields to test fixtures and API stub.  (It's foreseen that deriving/calling the Cospas-Sarsat API to get the derived data will be done in our Service)

## Guidance to review

- I've stubbed the derived data fields as "<Field name> to be derived from Hex ID" rather than stubbing "EPIRB", for example.  I did this because there's uncertainty around _if/how/when_ we'll be able to derive this stuff and I don't want MCA to think we already are deriving it.
- The API currently doesn't return the fields I've stubbed, so when deployed to dev/staging/prod this will display as "NO DATA ENTERED" or similar.  I could submit a slim PR to create those attributes on the API with just a string "<Field name> to be derived from Hex ID" as a more accurate representation of how the system might look in future?  But there's still uncertainty over even whether these fields are valuable, so I'm hesitant to do the work.
- Merging this completes the `Edit beacon summary` card (for now...)

## Things to check

- [ ] Environment variables have been updated

### CODE REVIEW

File: src/panels/beaconSummaryPanel/BeaconSummaryEditing.test.tsx [hunk 1]
```diff
@@ -19,9 +19,6 @@ describe("BeaconSummaryEditing", () => {
     expect(
       await screen.findByDisplayValue(beaconFixture.model as string)
     ).toBeVisible();
-    expect(
-      await screen.findByDisplayValue(beaconFixture.type as string)
-    ).toBeVisible();
```
Comment (zackads) [inline src/panels/beaconSummaryPanel/BeaconSummaryEditing.test.tsx:24]: `beaconFixture.type` is no longer editable, so remove the assertion that expects to find by the form input's current display value.

File: src/panels/beaconSummaryPanel/BeaconSummaryEditing.test.tsx [hunk 2]
```diff
@@ -98,4 +95,23 @@ describe("BeaconSummaryEditing", () => {
...
+
+    expect(
+      await screen.findByText(beaconFixture.protocolCode, { exact: false })
+    ).toBeVisible();
+    expect(
+      await screen.findByText(beaconFixture.codingMethod, { exact: false })
+    ).toBeVisible();
+    expect(screen.queryByDisplayValue(beaconFixture.protocolCode)).toBeNull();
+    expect(screen.queryByDisplayValue(beaconFixture.codingMethod)).toBeNull();
+  });
```
Comment (zackads) [inline src/panels/beaconSummaryPanel/BeaconSummaryEditing.test.tsx:116]: "expect to see the text on the page, but don't expect to see a form input with the same display value"

File: src/panels/beaconSummaryPanel/BeaconSummaryEditing.tsx [hunk 3]
```diff
@@ -153,6 +123,43 @@ export const BeaconSummaryEditing: FunctionComponent<{
...
+                              <b>Protocol code</b>, <b>coding method</b> and{" "}
+                              <b>beacon type</b> are automatically derived from
+                              the HEX ID. If you have identified a coding issue,
+                              please flag this record as 'Incorrectly Encoded'
+                              by clicking on the 'Quick Actions' button in the
+                              top right. You can also add any notes in the
+                              'Notes' tab below.
+                            </Typography>
+                          </Box>
+                        </Paper>
```
Comment (zackads) [inline src/panels/beaconSummaryPanel/BeaconSummaryEditing.tsx:159]: Option in future to extract this to a reusable `<InfoText>` component if this pattern sticks.

image

File: src/panels/beaconSummaryPanel/BeaconSummaryPanel.test.tsx [hunk 4]
```diff
@@ -29,19 +29,6 @@ describe("BeaconSummaryPanel", () => {
...
-      <BeaconSummaryPanel
-        beaconsGateway={beaconsGatewayDouble}
-        beaconId={beaconFixture.id}
-      />
-    );
-
-    expect(
-      await screen.findByText(beaconFixture.protocolCode as string)
-    ).toBeVisible();
-  });
```
Comment (zackads) [inline src/panels/beaconSummaryPanel/BeaconSummaryPanel.test.tsx:43]: Deleted test, as this behaviour is tested by the child viewing state.

**Review by tm1303 — APPROVED** [review]
looks good, nothing controversial here 🌟

**Pull request merged**